src/main/java/net/fabricmc/loader/discovery/ModResolver.java

```diff
@@ -455,6 +460,8 @@ protected void compute() {
 					rootDir = jarFs.get().getRootDirectories().iterator().next();
 				} catch (IOException e) {
 					throw new RuntimeException("Failed to open mod JAR at " + path + "!");
+				} catch (ZipError e) {
```

Comment:
any example of crash of zip error before?

Reply:
[here](https://paste.ee/p/2iyZZ)'s one, as well as in #185

Reply:
Yes, this is a good idea.

Reply:
No option to make this into a multi-cache and treat both the same?

Reply:
*catch

Reply:
It could be a multi-catch, but I feel like having different messages for different errors is better.